Engine/Engine/Core/InputPoller.h

```diff
@@ -18,15 +16,15 @@ class Input
 	static double mouseX, mouseY;
 	
 public:
-	Input(GLFWwindow* window);
+	void Init(GLFWwindow* window); // initializer takes a pointer to the window from the renderer class
 	inline double GetX() { return mouseX; }
 	inline double GetY() { return mouseY; }
-	void Init(GLFWwindow* window); // initializer takes a pointer to the window from the renderer class
+	static inline int GetKey(int key) { return m_keys[key]; }
 
 private:
 	friend static void key_callback(GLFWwindow* , int, int, int , int); // callback for keys
 	friend static void mouse_button_callback(GLFWwindow* , int , int , int);//callback for mousebutton
 	friend static void cursor_position_callback(GLFWwindow*, double, double);//callback for mouse cursor
```

Comment:
These "friend static" functions also do not compile on Linux. Removing the friend keyword fixes it. Additionally, the corresponding functions in InputPoller.cpp should be like "Input::key_callback"

Reply:
Removed friend keyword and adjusted .cpp accordingly.